IA/IA_consume_logs.py

```diff
@@ -0,0 +1,143 @@
+import os
+import argparse
+import json
+import logging
+import time
+import requests
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '-g',
+    '--guid',
+    help='This is the GUID of the target node on the OSF'
+)
+parser.add_argument(
+    '-d',
+    '--directory',
+    help='This is the target Directory for the project and its files'
+)
+
+parser.add_argument(
+    '-p',
+    '--pagesize',
+    help='How many logs should appear per file? Default is 100'
+)
+
+parser.add_argument(
+    '-t',
+    '--token',
+    help='Auth token for osf. This is required.'
+)
+
+
+BASE_URL = 'https://api.osf.io/'
+LOGS_URL = 'v2/registrations/{}/logs/?page[size]={}'
+
+def json_with_pagination(path, guid, page, url, token):
+    # Get JSON of registration logs
+    response = make_json_api_request(url, token)
+
+    # Craft filename based on page number
+    json_filename = guid + '-' + str(page) + '.json'
+    file_location = os.path.join(path, json_filename)
+    json_data = response['data']
+    with open(file_location, 'w') as file:
+        json.dump(json_data, file)
+    return response
+
+def make_json_api_request(url,token):
+
+    auth_header = { 'authorizataion' : 'Bearer ' + token}
+    keep_trying = True
+    response = None
+
+    while keep_trying:
+        keep_trying = False
+        try:
+            response = requests.get(url, headers=auth_header)
+            if response.status_code == 429:
+                keep_trying = retry
+                response_headers = response.headers
+                wait_time = response_headers['Retry-After']
+                if keep_trying:
+                    logging.log(logging.INFO, 'Throttled: retrying in {wait_time}s')
+                    time.sleep(int(wait_time))
+                else:
+                    logging.log(logging.ERROR, 'Throttled. Please retry after {wait_time}s')
+            elif response.status_code >= 400:
+                status_code = response.status_code
+                content = getattr(response, 'content', None)
+                raise requests.exceptions.HTTPError(
+                    'Status code {}. {}'.format(status_code, content))
+        except requests.exceptions.RequestException as e:
+            logging.log(logging.ERROR,'HTTP Request failed: {}'.format(e))
+            raise
+    try:
+        return response.json()
+    except json.decoder.JSONDecodeError:
+        return None
+
+def main(default_args=True):
+    # Arg Parsing
+    if (default_args):
+        args = parser.parse_args(['--guid', 'default', '--directory', 'default'])
+    else:
+        args = parser.parse_args()
+    args = parser.parse_args()
```

Comment:
@corbinSanders `args` is just getting over ridden regardless of https://github.com/CenterForOpenScience/osf-pigeon/pull/6/files#diff-7f16600c2bf0bca1486363000494cdabR85-R88